### PR TITLE
User Sign-Up Feature

### PR DESCRIPTION
**Pull request by IMMANUEL5015**
User model successfully created
Authentication Route for Signing up users has been created.
The controller for creating new users has also been created.
JWT authentication works as expected.
Emails pass through a validation process to ensure that they follow the correct format.
The password and confirm-password field must match each other before the user can be created and authenticated.
Moreso, user's passwords are salted and hashed before being saved to the database.

The Login functionality will be tackled next.

However, there are some issues with this sign-up feature
1. The mongoose "unique" property is not working well, so users can sign up with the email address and username of an existing user.
2. I am currently unable to write the tests for the remaining part of the test coverage, as all my efforts have been an exercise in futility.

### CODE REVIEW

**Comment by IMMANUEL5015** [comment]
The "unique" property seems to be working now. It seems like it works when it feels like doing so.

File: server/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        res.status(201).json({
+            status: 'Success',
+            token,
+            data: {
+                user: newUser
+            }
+        });
+    }catch(err){
+        res.status(400).json({
```
Comment (ngozi-ekekwe) [inline server/controllers/authController.js]: The error can be anything, 409 (if the user already exists) or 500 if its a server error or even 400. return `err.status`  and `err.message` if it exists

File: server/controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            status: 'fail',
+            message: err
```
Comment (ngozi-ekekwe) [inline server/controllers/authController.js]: I like naming things fully if they are not too long. eg `err` should be `error`. It makes it easy for people to read

File: server/models/usermodel.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        type: String,
+        required: [true, 'Please provide us with your email address'],
+        unique: [true, 'Email is already being used by someone else'],
+        lowercase: true,
+        validate: [validator.isEmail, 'Please provide a valid email address']
+    },
+    photo: String,
+    password: {
+        type: String,
+        required: [true, 'Please provide a password'],
```
Comment (ngozi-ekekwe) [inline server/models/usermodel.js]: If its possible let the user know the minimum length is 8

File: tests/authController.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+let req, res, next;
+
+beforeEach(() => {
+    req = httpMocks.createRequest();
+    res = httpMocks.createResponse();
+    next = null;
+});
+
+describe("authController should a a new user to the database", () => {
```
Comment (ngozi-ekekwe) [inline tests/authController.test.js]: there's a typo here

Comment (ngozi-ekekwe) [reply tests/authController.test.js]: You might want to modify your test cases

e.g

`describe('Auth Controller API", () => {...`

**Comment by ngozi-ekekwe** [comment]
Looks good!. Thank you for the changes

**Pull request merged**